spec/core/AsyncExpectationSpec.js

```diff
@@ -178,40 +178,52 @@ describe('AsyncExpectation', function() {
       });
     });
 
-    it('builds its message correctly when negated', function() {
+  });
+
+  describe('#toBeRejectedWith', function() {
+    it('passes if the promise is resolved to the expected value', function() {
       jasmine.getEnv().requirePromises();
 
+      var actual = Promise.resolve({foo: 42});
       var addExpectationResult = jasmine.createSpy('addExpectationResult'),
-        expectation = jasmineUnderTest.AsyncExpectation.factory({
+        expectation = new jasmineUnderTest.AsyncExpectation({
           util: jasmineUnderTest.matchersUtil,
-          actual: Promise.resolve(true),
+          actual: actual,
           addExpectationResult: addExpectationResult
         });
 
-      return expectation.not.toBeResolvedTo(true).then(function() {
-        expect(addExpectationResult).toHaveBeenCalledWith(false,
-          jasmine.objectContaining({
-            passed: false,
-            message: 'Expected a promise not to be resolved to true.'
-          })
-        );
+      return expectation.toBeRejectedWith({foo: 42}).then(function() {
+        expect(addExpectationResult).toHaveBeenCalledWith(true, {
+          matcherName: 'toBeRejectedWith',
+          passed: true,
+          message: '',
+          error: undefined,
+          errorForStack: jasmine.any(Error),
+          actual: actual
+        });
       });
     });
 
-    it('supports custom equality testers', function() {
+    it('fails if the promise is resolved to a different value', function() {
       jasmine.getEnv().requirePromises();
 
+      var actual = Promise.resolve({foo: 17});
```

Comment:
If we're trying to check the value comparison here, this should also be a `Promise.reject`.